### ISSUE TITLE
self.myImage.pixels[i] = color(0,0) works sometimes, doesn't work sometimes. 

### ISSUE THREAD

**Issue by ShunyaWatanabe** [processing/processing#5038]
Hello. In my project, I make a noise pattern using the code below. I want to make the white background color transparent, so that when one noise pattern covers the other, they overlap and create different noise pattern.

The idea create random number of rectangles with width w and height h, which are colored with fill(0), and save it as a png. load the png pixels, and render the background(255) to transparent. These fill and background are defined in setup(). 

```
    def create(self):        #Creates grids
        #Creates rectangles and saves grids
        for row in range(self.w):  
            for col in range(self.h):
                if int(randint(0, 1)) == 1:
                    rect(row*self.r, col*self.r, self.r, self.r)
        temp = get(self.x, self.y, self.w*self.r, self.h*self.r)
        temp.save("grid" + str(self.n) + ".png")
        background(255)
        
        self.myImage = loadImage("grid" + str(self.n) + ".png")
        self.myImage.loadPixels()
        
        #Sets all black grid pixels to transparent
        for i in range(self.w*self.h*self.r*self.r):
            if self.myImage.pixels[i] == color(255):
                self.myImage.pixels[i] = color(0,0)

        self.myImage.updatePixels()

    def display(self):
        image(self.myImage, self.x, self.y)
```
This code sometimes works and gives me black noise patterns with transparent background so that I can see one noise pattern going over the other noise pattern. But sometimes it doesn't. With the same piece of code. Instead, it gives me myImage that is all black. Sometimes one of the two noise patterns becomes black, sometimes both, sometimes two of them become actual noise patterns. 

I also tried it with self.myImage.pixels[i] = color(0,0,0,0) and also the transparent color code (my friend used it and i forget the code). Transparent color code doesn't give me whole black picture but also doesn't render white color to anything.

I'm using Processing 3.3.2 and windows 64 bit, but it is also the case in mac os.

Any ideas why? or is this a bug?

Below is how it looks when it doesn't work well. Both noise patterns use the same code above.

![capture](https://cloud.githubusercontent.com/assets/25846438/25565461/2729add6-2dd8-11e7-81f1-73a8c2f1fbe3.PNG)

**Comment by benfry** [comment]
Unlikely to be a bug… perhaps check with forum.processing.org to see if you can get some coding help. But report back here if you find it's actually a bug in the Processing software.

**Comment by ShunyaWatanabe** [comment]
I was hoping for a constructive answer, benfry, but you didn't give me even a line of help. 
Anyways, I found the reason why my code has been acting like this, and I'd like to share it here.
![2017-05-01 8 38 05](https://cloud.githubusercontent.com/assets/25846438/25586170/3710f7b8-2eaf-11e7-9146-ba717dab2539.png)
Main reason was because of this. However, what it is written in the troubleshooting page is a little bit misleading. There are 2 things I'd like to mention.

1. "color(0,0,0,0) is black." is only half true in my case. My code works sometimes and color(0,0,0,0) gives me a transparent color. However, some of the times it does give me black too. 
2. The "ways to work around" only consists of situations when the code uses fill, so it is harder to realize the problem when one uses assignment to a pixel color, as in my case.

Maybe it helps to add a few more explanations in the troubleshooting page?

`self.myImage.pixels[i] = color(0,0,0,0)`
This is a piece of code from my work, and it gives me sometimes transparent and sometimes black. When I say sometimes, I mean within the for loop, they will give the same color but once it goes out of the for loop, they will maybe next time give the different color even though it's the same for loop at the same place in your code. These are three variations of the output with the same code. 
![2017-05-01 8 29 06](https://cloud.githubusercontent.com/assets/25846438/25586817/2e4f94d8-2eb2-11e7-974a-afb3a5eecdd1.png)
![2017-05-01 8 31 21](https://cloud.githubusercontent.com/assets/25846438/25586833/3956b2ee-2eb2-11e7-9e2d-954ac1e44348.png)
![2017-05-01 8 30 58](https://cloud.githubusercontent.com/assets/25846438/25586838/4100a4c8-2eb2-11e7-9f56-7b88bf1978ba.png)
 I used one of the ways to work around as it is mentioned at the troubleshooting page.
`self.myImage.pixels[i] = color(0,0,0,1)`
I really want the actual transparent color, but I guess I can't do that.
